inbm/dispatcher-agent/dispatcher/aota/application_command.py

```diff
@@ -135,27 +135,15 @@ def update(self) -> None:
             # Move driver to CentOS filesystem
             shutil.move(driver_path, driver_centos_path)
 
-            # Remove old driver
-            old_driver_name = self.identify_package(driver_path.split('/')[-1])
-            if not old_driver_name:
-                raise AotaError(
-                    f'AOTA Command Failed: Unsupported driver {driver_path.split("/")[-1]}')
-            uninstall_driver_cmd = DOCKER_CHROOT_PREFIX + \
-                f'/usr/bin/rpm -e --nodeps {old_driver_name}'
-            out, err, code = PseudoShellRunner().run(uninstall_driver_cmd)
-            logger.debug(out)
-            # If old packages wasn't install on system, it will return error too.
-            if code != 0 and "is not installed" not in str(err):
-                raise AotaError(err)
-
             chroot_driver_path = driver_centos_path.replace("/host", "")
-            install_driver_cmd = DOCKER_CHROOT_PREFIX + f'/usr/bin/rpm -ivh {chroot_driver_path}'
+            install_driver_cmd = CHROOT_PREFIX + \
+                f'/usr/bin/rpm -Uvh --force {chroot_driver_path}'
             logger.debug(f" Updating Driver {driver_path.split('/')[-1]} ...")
             out, err, code = PseudoShellRunner().run(install_driver_cmd)
             logger.debug(out)
             if code != 0:
                 raise AotaError(err)
-            self._reboot(DOCKER_CHROOT_PREFIX + '/usr/sbin/shutdown -r 0')
+            self._reboot(CHROOT_PREFIX + '/usr/sbin/shutdown -r 0')
```

Comment:
same here with DOCKER_CHROOT_PREFIX: works better for communicating with systemd

Reply:
@gblewis1[](https://user-images.githubusercontent.com/58315865/156096798-f4dbbe94-5dc4-4769-bd32-0bb95e1f3390.png) Using DOCKER_CHROOT_PREFIX, it failed to execute the rpm command. With CHROOT_PREFIX, the rpm command pass. Can we use the CHROOT_PREFIX first in CentOS driver installation and later debug the issue?

![image](https://user-images.githubusercontent.com/58315865/156485181-cbc03c17-0a6a-4544-930c-51c557cdc7d6.png)

Reply:
@yengliong93 Let's take a look at this on the host system.. both of these are with docker chroot (in contrast to the comment saying CHROOT_PREFIX there's a pass), and the first example that says 'fail' is rpm -e (erase), second one is rpm -Uvh (upgrade); both seem to be failing, in contrast to the marking on the screenshot.

Reply:
looks like CHROOT_PREFIX is correct here